### PR TITLE
Update timestamp publisher output path

### PR DESCRIPTION
**Pull request by jack-moseley**
Modify TimestampDataPublisher so that the output path is in the correct format.

Before:
`outputDir/timestamp/dbname_tablename_xxxxx`

After:
`outputDir/dbname.tablename/timestamp`

### CODE REVIEW

**Comment by jack-moseley** [comment]
@ydai1124 @chavdar please review

File: gobblin-core/src/main/java/gobblin/publisher/TimestampDataPublisher.java [hunk 1]
```diff
@@ -32,10 +33,51 @@ public TimestampDataPublisher(State state) throws IOException {
...
+   */
   @Override
-  protected Path getPublisherOutputDir(WorkUnitState workUnitState, int branchId) {
-    Path outputDir = super.getPublisherOutputDir(workUnitState, branchId);
-    outputDir = new Path(outputDir, timestamp);
-    return outputDir;
+  protected void movePath(ParallelRunner parallelRunner, State state, Path src, Path dst, int branchId)
+      throws IOException {
+
+    String outputDir = dst.toString().substring(0, dst.toString().lastIndexOf('/'));
```
Comment (ydai1124) [inline gobblin-core/src/main/java/gobblin/publisher/TimestampDataPublisher.java]: You can use `dst.getName()` and `dst.getParent().getName()` to replace the parsing of `/` here.

File: gobblin-core/src/main/java/gobblin/publisher/TimestampDataPublisher.java [hunk 2]
```diff
@@ -32,10 +33,51 @@ public TimestampDataPublisher(State state) throws IOException {
...
+    super.movePath(parallelRunner, state, src, newDst, branchId);
+  }
+
+  /**
+   * Translate schema name to "dbname.tablename" to use in path
+   *
+   * @param schemaName In format "dbname_tablename_xxxxx"
+   * @return db and table name in format "dbname.tablename"
+   */
+  private String getDbTableName(String schemaName) {
```
Comment (ydai1124) [inline gobblin-core/src/main/java/gobblin/publisher/TimestampDataPublisher.java]: Can you add a precondition check here to enforce the expected format?

File: gobblin-core/src/main/java/gobblin/publisher/TimestampDataPublisher.java (hunk 1 above)
Comment (jack-moseley) [inline gobblin-core/src/main/java/gobblin/publisher/TimestampDataPublisher.java]: fixed

File: gobblin-core/src/main/java/gobblin/publisher/TimestampDataPublisher.java (hunk 2 above)
Comment (jack-moseley) [inline gobblin-core/src/main/java/gobblin/publisher/TimestampDataPublisher.java]: Added a regex check

File: gobblin-core/src/main/java/gobblin/publisher/TimestampDataPublisher.java [hunk 3]
```diff
@@ -13,15 +13,17 @@
...
+import com.google.common.base.Preconditions;
+import gobblin.util.ParallelRunner;
+import gobblin.util.WriterUtils;
 import gobblin.configuration.State;
 import gobblin.configuration.WorkUnitState;
 
 
 /**
- * Write output to {finaldir}/{currentTimestamp}
+ * Write output to {finaldir}/{dbname.tablename}/{currenttimestamp}
```
Comment (chavdar) [inline gobblin-core/src/main/java/gobblin/publisher/TimestampDataPublisher.java]: Please have a more detailed description of:
- Preconditions: what is the writer output expected to be
- Post conditions: what does published output look like

Comment (jack-moseley) [inline gobblin-core/src/main/java/gobblin/publisher/TimestampDataPublisher.java]: updated javadoc

**Pull request merged**